lib/cli-engine.js

```diff
@@ -300,18 +300,31 @@ function processFile(filename, configHelper, options) {
 
 /**
  * Returns result with warning by ignore settings
- * @param {string} filePath File path of checked code
- * @returns {Result} Result with single warning
+ * @param {string} filePath - File path of checked code
+ * @param {string} baseDir  - Absolute path of base directory
+ * @returns {Result}           Result with single warning
  * @private
  */
-function createIgnoreResult(filePath) {
+function createIgnoreResult(filePath, baseDir) {
+    var message;
+
+    if (/^\./.test(path.basename(filePath))) {
+        message = "Hidden file ignored by default.  Use '--ignore-pattern !.*' to override.";
```

Comment:
You can try it on your branch with:

```
eslint node_modules/escope/gulpfile.js --ignore-pattern '!node_modules/escope/gulpfile.js'
```

vs:
```
eslint node_modules/escope/gulpfile.js --ignore-pattern '!node_modules/*'
```

Also, bash needs single, not double quotes with "!" for some reason. :(